bobcat/bb-aem-touch-ui/src/main/java/com/cognifide/qa/bb/aem/touch/pageobjects/touchui/ConfigDialog.java

```diff
@@ -99,14 +100,35 @@ public void configureWith(ComponentConfiguration config) {
     confirm();
   }
 
-  private void switchTab(String tabLabel) {
+  public DialogField getFieldOnTab(String label, String tab, String fieldType) {
+    switchTab(tab);
+    return getField(label, fieldType);
+  }
+
+  public ConfigDialog setField(String label, String fieldType, Object value) {
+    getField(label, fieldType).setValue(value);
+    return this;
+  }
+
+  public DialogField getField(String label, String fieldType) {
+    WebElement parent;
+    if (tabs.isEmpty()) {
+      parent = dialog;
+    } else {
+      parent = activeTab;
+    }
+    return dialogConfigurer.getDialogField(parent, label, fieldType);
+  }
+
+  public ConfigDialog switchTab(String tabLabel) {
     if (!tabs.isEmpty()) {
```

Comment:
We should throw an exception here, e.g. `IllegalStateException` - if someone explicitly invokes this method, he is expecting tabs in the dialog.